cosign.py

```diff
@@ -334,10 +336,16 @@ def sign_image(
     )
 
     manifest_bytes = json.dumps(signed_manifest.as_dict()).encode('utf-8')
-
-    oci_client.put_manifest(
-        image_reference=signature_image_reference,
-        manifest=manifest_bytes,
-    )
+    manifest_digest = f'sha256:{hashlib.sha256(manifest_bytes).hexdigest()}'
+    manifest_digest_ref = f'{signature_image_reference.ref_without_tag}@{manifest_digest}'
+
```

Comment:
nit: maybe add a brief comment (e.g. "re-uploading artefact w/ same digest is a no-op") or similar)